plugins/image/src/upload.component.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react'
+//@ts-ignore
+import { Uploader, UploadField } from '@navjobs/upload'
+import { UploadProps } from './types'
+import Text from '@splish-me/editor-ui/lib/sidebar-elements/sidebartext'
+
+enum FileError {
+  TOO_MANY_FILES,
+  NO_FILE_SELECTED,
+  BAD_EXTENSION,
+  FILE_TOO_BIG,
+  UPLOAD_FAILED
+}
+
+export class Upload extends React.Component<UploadProps> {
+  matchesAllowedExtensions(fileName: string) {
```

Comment:
This looks like it could be solved in a simpler way ^^'. Can't you just get the file extension and check if it is an element of `this.props.config.allowedExtensions`?

Reply:
yeah probably. Got this from the ory editor plugin (and I think it corresponds to this answer here: https://stackoverflow.com/a/17355937) but we could also do it the other way around without regex.